### PR TITLE
Add E2E tests for OSM

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
Signed-off-by: Waleed Malik <ahmedwaleedmalik@gmail.com>

**What does this PR do / Why do we need it**:
This PR introduces a very basic E2E test for OSM. We've used AWS and k8s v1.24 for now, it's sufficient IMO since we only want to test the integration at the moment.

**Does this PR close any issues?**:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: hack/ci/testdata/kubermatic.yaml
Comment (xrstf) [inline hack/ci/testdata/kubermatic.yaml]: This would enable the feature flag for all e2e tests. What effects will this have?

Comment (ahmedwaleedmalik) [reply hack/ci/testdata/kubermatic.yaml]: It will just install the CRDs and RBAC for seed. We'll make it enabled by default in a few days but for now I have made this a dynamically populated field.

**Comment by ahmedwaleedmalik** [comment]
/hold
We need the updated image tag after merging the following PRs to proceed with this one:
- https://github.com/kubermatic/operating-system-manager/pull/184
- https://github.com/kubermatic/operating-system-manager/pull/181

**Review by xrstf — APPROVED** [review]
/approve

**Pull request merged**
